packages/flutter/test/painting/image_provider_network_image_test.dart

```diff
@@ -70,6 +70,42 @@ void main() {
     expect(httpClient.request.response.drained, true);
   }, skip: isBrowser); // [intended] Browser implementation does not use HTTP client but an <img> tag.
 
+  test('Expect thrown exception with statusCode - evicts from cache and drains, when using ResizeImage', () async {
+    const int errorStatusCode = HttpStatus.notFound;
+    const String requestUrl = 'foo-url';
+
+    httpClient.request.response.statusCode = errorStatusCode;
+
+    final Completer<dynamic> caughtError = Completer<dynamic>();
+
+    final ImageProvider imageProvider = ResizeImage(NetworkImage(nonconst(requestUrl)), width: 5, height: 5);
+    expect(imageCache.pendingImageCount, 0);
+    expect(imageCache.statusForKey(imageProvider).untracked, true);
+
+    final ImageStream result = imageProvider.resolve(ImageConfiguration.empty);
+
+    expect(imageCache.pendingImageCount, 1);
+
+    result.addListener(ImageStreamListener((ImageInfo info, bool syncCall) {
+    }, onError: (dynamic error, StackTrace? stackTrace) {
+      caughtError.complete(error);
+    }));
+
+    final dynamic err = await caughtError.future;
```

Comment:
Please don't use dynamic. Use `Object` or `Object?`

Reply:
I copy-paste-modified it from the existing test https://github.com/fzyzcjy/flutter/blob/ccac069c5bc4392e6c23281f2432b50b3f7e6d59/packages/flutter/test/painting/image_provider_network_image_test.dart#L59. Shall I change that as well?